src/lib.rs

```diff
@@ -126,7 +142,30 @@ impl Uart {
         }
     }
 
-    /// TODO: implement init
+    /// Initializes PL011 UART.
+    /// clk: Uart clock in Hz.
```

Comment:
I guess you mean `clock` rather than clk. What clock rate is the caller supposed to pass here?

Please also document the `baud_rate` parameter.